Error on running the website locally using newer Hugo version

**This is a Bug Report**

<!-- Thanks for filing an issue! Before submitting, please fill in the following information. -->
<!-- See https://kubernetes.io/docs/contribute/start/ for guidance on writing an actionable issue description. -->

<!--Required Information-->
**Problem:**

I followed the section "Running the website locally using Hugo" in README of this repo, but I met an error.

With
- Ubuntu 18.04 desktop amd64
- the latest commit of this repo (0ea4666)
- Docsy whose version is the same as of https://github.com/kubernetes/website/tree/main/themes (docsy @ 1c77bb2)
- npm version: 6.14.15
- node version: v14.18.1
- go version go1.17.3 linux/amd64
- hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended linux/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=mage
- Docker Client/Server: 20.10.11

---

[How to reproduce]

```
❯ git submodule update --init --recursive --depth 1
# OK with no output
```

```
❯ npm ci
npm WARN prepare removing existing node_modules/ before installation

> core-js-pure@3.6.5 postinstall /home/jhseo/k8s-website/node_modules/core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

added 120 packages in 1.414s
```

```
❯ make serve
hugo server --buildFuture --environment development
Start building sites … 
hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended linux/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=mage
WARN 2022/04/18 15:43:16 The google_news internal template will be removed in a future release. Please remove calls to this template. See https://github.com/gohugoio/hugo/issues/9172 for additional information.
WARN 2022/04/18 15:43:17 .Path when the page is backed by a file is deprecated and will be removed in a future release. We plan to use Path for a canonical source path and you probably want to check the source is a file. To get the current behaviour, you can use a construct similar to the one below:

  {{ $path := "" }}
  {{ with .File }}
        {{ $path = .Path }}
  {{ else }}
        {{ $path = .Path }}
  {{ end }}


Re-run Hugo with the flag --panicOnWarning to get a better error message.
Error: Error building site: "/home/jhseo/k8s-website/content/en/docs/concepts/workloads/controllers/statefulset.md:410:12": failed to render shortcode "api-reference": failed to process shortcode: "/home/jhseo/k8s-website/layouts/shortcodes/api-reference.html:7:17": execute of template failed: template: shortcodes/api-reference.html:7:17: executing "shortcodes/api-reference.html" at <$page.URL>: can't evaluate field URL in type page.Page
Built in 2752 ms
Makefile:54: recipe for target 'serve' failed
make: *** [serve] Error 255
```

**Proposed Solution:**

--- a/layouts/shortcodes/api-reference.html
+++ b/layouts/shortcodes/api-reference.html
@@ -4,4 +4,4 @@
 {{ $textArg := .Get "text" }}
 {{ $page := site.GetPage "page" (printf "%s/%s" $base $pageArg) }}
 {{ $metadata := $page.Params.api_metadata }}
-<a href="{{ $page.URL }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>
+<a href="{{ $page.RelPermalink }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>

